client/dashboard/task-list/tasks/shipping/index.js

```diff
@@ -0,0 +1,211 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { filter } from 'lodash';
+import { withDispatch } from '@wordpress/data';
+
+/**
+ * WooCommerce dependencies
+ */
+import { Card, Stepper } from '@woocommerce/components';
+import { getHistory, getNewPath } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import StoreLocation from './location';
+import ShippingRates from './rates';
+import withSelect from 'wc-api/with-select';
+
+class Shipping extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.initialState = {
+			isPending: false,
+			step: 'store_location',
+			shippingZones: [],
+		};
+
+		this.state = this.initialState;
+
+		this.completeStep = this.completeStep.bind( this );
+	}
+
+	componentDidMount() {
+		this.reset();
+	}
+
+	reset() {
+		this.setState( this.initialState );
+	}
+
+	async fetchShippingZones() {
+		this.setState( { isPending: true } );
+		const { countryCode, countryName } = this.props;
+		const shippingZones = [];
+		const zones = await apiFetch( { path: '/wc/v3/shipping/zones' } );
+		let hasCountryZone = false;
+
+		await Promise.all(
+			zones.map( async zone => {
+				// "Rest of the world zone"
+				if ( 0 === zone.id ) {
+					zone.methods = await apiFetch( { path: `/wc/v3/shipping/zones/${ zone.id }/methods` } );
+					zone.name = __( 'Rest of the world', 'woocommerce-admin' );
+					zone.toggleEnabled = true;
+					shippingZones.push( zone );
+					return;
+				}
+
+				// Return any zone with a single location matching the country zone.
+				zone.locations = await apiFetch( { path: `/wc/v3/shipping/zones/${ zone.id }/locations` } );
+				if ( 1 === zone.locations.length && countryCode === zone.locations[ 0 ].code ) {
```

Comment:
Would we consider relaxing this a bit and just checking for the store's country to be in the zone?

In my testing I had a zone for USA+Canada, and a new US-only zone was created.

Reply:
I'm okay to allow multiple countries if you are.  The original designs call for a single country with a single flat rate method, but this opens a can of worms for pre-existing methods if they visit the step after completion.

I think with the current refactor, it's okay to allow zones with multiple locations since we can show the flag for each place.  Also curious if we should note on this step that this applies only to flat rate methods in case a user has another method not shown here. /cc @jameskoster

<img width="414" alt="Screen Shot 2019-08-08 at 1 55 00 PM" src="https://user-images.githubusercontent.com/10561050/62682279-5fac6780-b9ee-11e9-838e-fc8ef5dd1a6b.png">

Reply:
I defer to design on this one.

> this opens a can of worms for pre-existing methods if they visit the step after completion

Perhaps it's better to have a future issue to handle this case a little more elegantly?

It's probably more trouble than its worth to look for zones containing the store's country and only a flat rate shipping method at this time.